updated issue templates and added pull_request templates

iamrajiv opened the pull request:
Fixes: #9342

iamrajiv commented:
@gvanrossum please take a look

JelleZijlstra commented on .github/issue_template/bug.md:
```diff
@@ -0,0 +1,56 @@
+
+(A clear and concise description of what the bug is.)
+
+### Have you read the [Contributing Guidelines](https://github.com/python/mypy/blob/master/CONTRIBUTING.md)?
```
Why should people who report a bug read the contributing guidelines?

iamrajiv replied:
@JelleZijlstra I have removed the contributing guidelines link from all the issue templates.

JelleZijlstra commented on .github/issue_template/bug.md:
```diff
@@ -0,0 +1,56 @@
+1. Step 2...
+1. Step 3...
+
+## Expected Behavior
```
I'm not sure the "Expected behavior" and "Actual behavior" sections are useful. A lot of the time they will just be "Expected behavior: No mypy error", "Actual behavior: Got a mypy error".

iamrajiv replied:
@JelleZijlstra I have added this message at end of the `bug.md` file
```
(You can freely edit this text, please remove all the lines
you believe are unnecessary.)
```

JelleZijlstra commented on .github/issue_template/feature.md:
```diff
@@ -0,0 +1,21 @@
+
+(Write your answer here.)
+
+## Motivation
```
I'm not sure I understand the difference between "Motivation" and "Pitch".

iamrajiv replied:
@JelleZijlstra I have removed the Motivation section.

JelleZijlstra commented on .github/issue_template/proposal.md:
```diff
@@ -0,0 +1,13 @@
+---
+name: 💥 Proposal
```
How is this different from "Feature"?

iamrajiv replied:
@JelleZijlstra I have deleted the `Proposal.md` file.

gvanrossum commented on .github/issue_template/bug.md:
```diff
@@ -0,0 +1,55 @@
+<!--
+  How did you expect your project to behave?
+  It’s fine if you’re not sure your understanding is correct.
+  Write down what you thought would happen.
```
Maybe just add "If you just expected no errors, you can delete this section."

gvanrossum commented on .github/issue_template/bug.md:
```diff
@@ -0,0 +1,55 @@
+<!--
+  Did something go wrong?
+  Is something broken, or not behaving as you expected?
+  Describe this section in detail, and attach screenshots if possible.
```
I don't think for mypy screenshots are ever relevant. We don't want people to attach screenshots of error messages (that's quite counterproductive).

gvanrossum commented on .github/issue_template/bug.md:
```diff
@@ -0,0 +1,55 @@
+- Python version used:
+- Operating system and version:
+
+(You can freely edit this text, please remove all the lines
+you believe are unnecessary.)
```
Make this another HTML comment, otherwise we'll see it in most issues...

gvanrossum commented on .github/issue_template/bug.md:
```diff
@@ -0,0 +1,55 @@
+
+<!-- Include as many relevant details about the environment you experienced the bug in -->
+
+- Mypy version used:
```
Add mypy config flags, which are very important:
```suggestion
- Mypy version used:
- Mypy command line flags:
- Mypy  configuration options from `mypy.ini` (and other config files):
```

gvanrossum commented on .github/issue_template/documentation.md:
```diff
@@ -0,0 +1,9 @@
+---
+name: 📚 Documentation
+about: Report an issue related to documentation
```
Maybe "Report a problem with the documentation". But again, might as well leave blank.

gvanrossum commented on .github/issue_template/documentation.md:
```diff
@@ -0,0 +1,9 @@
+
+## 📚 Documentation
+
+(A clear and concise description of what the issue is.)
```
```suggestion
(A clear and concise description of the issue.)
```

gvanrossum commented on .github/issue_template/feature.md:
```diff
@@ -0,0 +1,13 @@
+
+## 🚀 Feature
+
+(A clear and concise description of what the feature is.)
```
```suggestion
(A clear and concise description of your feature proposal.)
```

gvanrossum commented on .github/issue_template/feature.md:
```diff
@@ -0,0 +1,13 @@
+
+## Pitch
+
+(Please explain why this feature should be implemented and how it would be used.)
```
Maybe add an encouragement to add examples.

gvanrossum commented on .github/issue_template/feature.md:
```diff
@@ -0,0 +1,13 @@
+---
+name: 🚀 Feature
+about: Submit a proposal for a new feature
```
```suggestion
about: Submit a proposal for a new mypy feature
```
Also, I wonder if we should add somewhere that this is *not* the place to propose new type system features -- those should generally be discussed on `typing-sig@python.org` or in https://github.com/python/typing/issues and probably have to go through the PEP process (see https://www.python.org/dev/peps/pep-0001/).

gvanrossum commented on .github/issue_template/question.md:
```diff
@@ -0,0 +1,14 @@
+
+## ❓ Questions and Help
+
+### Please note that this issue tracker is not a help form and this issue will be closed.
```
Yes! Thank you!

gvanrossum commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,25 @@
+<!--
+Thank you for sending the PR! We appreciate you spending the time to work on these changes.
```
I wonder if we need this much happy talk. The text box is only 15 lines tall on my computer, and that means that the `Description` header is the bottom line visible without scrolling. Honestly I would drop this entire comment and just start with "Have you read ..."

gvanrossum commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,25 @@
+
+### Have you read the [Contributing Guidelines](https://github.com/python/mypy/blob/master/CONTRIBUTING.md)?
+
+(Write your answer here.)
```
```suggestion
(Once you have, delete this section. If you leave it in, your PR may be closed without action.)
```

gvanrossum commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,25 @@
+
+### Description
+
+(Write your answer here.)
```
```suggestion
(Explain how this PR changes mypy.)
```
And honestly if there are any related issues or PRs I'd like to see them listed here, not in a separate section at the bottom. Ideally in the form "Closes #NNN" or "Fixes #NNN" (we should explicitly mention these, many people don't know about those!)

gvanrossum commented on .github/pull_request_template.md:
```diff
@@ -0,0 +1,25 @@
+
+## Test Plan
+
+(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)
```
I don't think so. We have automated tests. Every PR that changes code should have tests (unless it's a pure refactoring that doesn't change any external behavior -- but we're not so keen on those). So we should provide specific, actionable instructions, e.g.

- If this is a documentation change, rebuild the docs (_link to instructions_) and review the changed pages for markup errors.
- If this is a code change, include new tests (_link to the testing docs_). Be sure to run the tests locally and fix any errors before submitting the PR (_more instructions_).
- If this change cannot be tested by the CI, please explain how to verify it manually.

iamrajiv commented:
> I tried this in https://github.com/gvanrossum/mypy/issues and the PR template shows fine, but the issue templates don't. What am I doing wrong?

@gvanrossum I think issue templates does not work fine because the folder is named as `issue_template` but it should be named as `ISSUE_TEMPLATE`.

iamrajiv commented:
Thank you for suggestions @gvanrossum I have updated the templates. Also, I have removed `about` section from issue templates because I think that is not so relevant.

gvanrossum commented:
Sorry, the issues template does not work. I am rolling this back.
